VAMobile/src/api/notifications/loadPushPreferences.tsx

```diff
@@ -0,0 +1,43 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+import { useQuery } from '@tanstack/react-query'
+
+import queryClient from 'api/queryClient'
+import { GetPushPrefsResponse, LoadPushPreferencesData } from 'api/types'
+import { get } from 'store/api'
+import { notificationsEnabled } from 'utils/notifications'
+
+import { notificationKeys } from './queryKeys'
+import { DEVICE_ENDPOINT_SID } from './registerDevice'
+
+/**
+ * Fetch user push preferences
+ */
+const loadPushPreferences = async (): Promise<LoadPushPreferencesData | undefined> => {
+  const systemNotificationsOn = await notificationsEnabled()
+  const endpoint_sid = await AsyncStorage.getItem(DEVICE_ENDPOINT_SID)
+  if (!endpoint_sid) return
+  const response = await get<GetPushPrefsResponse>(`/v0/push/prefs/${endpoint_sid}`)
+  const previousData = queryClient.getQueryData(notificationKeys.settings) as LoadPushPreferencesData
+  return {
+    preferences: response?.data.attributes.preferences || [],
+    systemNotificationsOn: systemNotificationsOn,
+    deviceToken: previousData?.deviceToken,
+    initialUrl: previousData?.initialUrl,
+    tappedForegroundNotification: previousData?.tappedForegroundNotification || false,
```

Comment:
For this one I don't think this would be an issue as the developer screen would mean we are in demo mode and that means the api isn't hit unnecessarily as we intercept it. So I believe these 3 are fine to leave as is

Reply:
updated